backend/CircleForms/Controllers/V1/PostsController.cs

```diff
@@ -181,22 +207,37 @@ public async Task<IActionResult> Post(PostRequestContract postContract)
     [ProducesResponseType(typeof(PostResponseContract), StatusCodes.Status200OK, "application/json")]
     [ProducesResponseType(typeof(PostDetailedResponseContract), StatusCodes.Status200OK, "application/json")]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
-    public async Task<IActionResult> GetDetailed(string id)
+    public async Task<IActionResult> GetDetailed(string id, [FromQuery] string key = "")
     {
         var claim = HttpContext.User.Identity?.Name;
-        var post = await _postRepository.Get(id);
+        var post = await _postRepository.GetCached(id);
         if (post is null)
         {
             return NotFound();
         }
 
+        Post forceRequestedPost = null;
+        if (post.Accessibility == Accessibility.Link)
+        {
+            if (string.IsNullOrEmpty(key) || key.Length != 6)
```

Comment:
This will probably need an anti-bruteforce logic here later

Reply:
Also, do you think we'd want link-only posts to require auth?

Reply:
>Also, do you think we'd want link-only posts to require auth?
then we should do this in another route, probably something like /posts/{id}/private?key={key}